### ISSUE TITLE
systemd: no directory /run/fail2ban to contain the socket file

### ISSUE THREAD

**Issue by khumarahn** [fail2ban/fail2ban#1384]
Hi,

I tried starting fail2ban on gentoo with systemd. It does not start with the following error in the system journal:

```
ERROR  There is no directory /run/fail2ban to contain the socket file /run/fail2ban/fail2ban.sock.
```

I fixed the problem brutally by `mkdir /run/fail2ban`. It worked then just fine. Is

```
ExecStartPre=/bin/mkdir -p /run/fail2ban
```

forgotten in the service file? Or it is supposed to be done differently?

Sorry if this question has been asked. I tried searching, but had no luck. Also, I am a noob.

**Comment by khumarahn** [comment]
It turns out that gentoo installs the tmpfiles.d file, but for me the directory wasn't created because I did not reboot, and tmpfiles.d did not refresh on its own.

It looks now that a proper solution would be to add either
```ExecStartPre=/bin/mkdir -p /run/fail2ban```
or
```RuntimeDirectory=fail2ban```

And probably it is time to migrate from /var/run to /run everywhere.

**Comment by khumarahn** [comment]
https://github.com/fail2ban/fail2ban/pull/1390

**Comment by sebres** [comment]
I'll close this, because it was fixed in #1618 for systemd service... (for 0.10 branch, can deviate by 0.9).
But I believe normally it is an issue of the maintainers of corresponding distribution...
We can not (and want not really) take care of all possible systems in *nix- and linux-zoo in frame of fail2ban development...

The way with `RuntimeDirectory=fail2ban` (from #1390) looks good and IMHO should be validated by @fail2ban/maintainers.

**Comment by thomasvincent** [comment]
Fixed in fail2ban:0.10

**Comment by lukapaunovic** [comment]
In CentOS 7 Minimal for ARM (raspberry Pi) i encountered this issue with latest fail2ban. 
Updating service file `/run/systemd/generator.late` (which is the only path where i found fail2ban.service) didn't help, because file would get restored as soon I restarted the service.
So, i added this to crontab instead

`@reboot /bin/mkdir -p /var/run/fail2ban`

Because folder disappears every time I reboot my system.
I am not sure why or how is this happening, but if someone knows please do write.
